### PR TITLE
jenkinsfiles: remove unused environment variables

### PR DESCRIPTION
**Pull request by aanm**
As we are no longer building images in ginkgo, we can remove these
environment variables.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/15141

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 1]
```diff
@@ -83,6 +71,7 @@ pipeline {
                     }
                     if (env.RACE?.trim()) {
```
Comment (nebril) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile]: `s/env.RACE?.trim()/env.run_with_race_detection?.trim()` , `RACE` was set at line 23

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 2]
```diff
@@ -100,6 +88,7 @@ pipeline {
                     }
                     if (env.RACE?.trim()) {
```
Comment (nebril) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: same as gke jenkinsfile

File: jenkinsfiles/ginkgo-gke.Jenkinsfile [hunk 3]
```diff
@@ -83,6 +71,7 @@ pipeline {
...
                         env.DOCKER_TAG = env.DOCKER_TAG + "-race"
+                        env.RACE = "true"
```
Comment (nebril) [inline jenkinsfiles/ginkgo-gke.Jenkinsfile]: This doesn't do anything, let's get rid of `RACE` env var in all jenkinsfiles aside from runtime.

File: jenkinsfiles/ginkgo-kernel.Jenkinsfile [hunk 4]
```diff
@@ -100,6 +88,7 @@ pipeline {
...
                         env.DOCKER_TAG = env.DOCKER_TAG + "-race"
+                        env.RACE = "true"
```
Comment (nebril) [inline jenkinsfiles/ginkgo-kernel.Jenkinsfile]: same as gke jenkinsfile

File: jenkinsfiles/ginkgo-runtime-kernel.Jenkinsfile [hunk 5]
```diff
@@ -28,14 +28,6 @@ pipeline {
                 returnStdout: true,
                 script: 'if [ "${run_with_race_detection}" = "" ]; then echo -n ""; else echo -n "1"; fi'
             )}"""
-        LOCKDEBUG="""${sh(
```
Comment (nebril) [inline jenkinsfiles/ginkgo-runtime-kernel.Jenkinsfile]: I think we want these env vars to be set, as they are passed to runtime VM which causes them to be passed to `make` targets run inside the vm which build the binaries.

File: jenkinsfiles/ginkgo-runtime-kernel.Jenkinsfile [hunk 6]
```diff
@@ -28,14 +28,6 @@ pipeline {
...
-                returnStdout: true,
-                script: 'if [ "${run_with_race_detection}" = "" ]; then echo -n ""; else echo -n "1"; fi'
-            )}"""
-        BASE_IMAGE="""${sh(
```
Comment (nebril) [inline jenkinsfiles/ginkgo-runtime-kernel.Jenkinsfile]: ditto

File: jenkinsfiles/kubernetes-upstream.Jenkinsfile [hunk 7]
```diff
@@ -87,6 +75,7 @@ pipeline {
                         env.DOCKER_TAG = env.ghprbActualCommit
                     } else {
                         env.DOCKER_TAG = env.GIT_COMMIT
+                        env.RACE = "true"
```
Comment (nebril) [inline jenkinsfiles/kubernetes-upstream.Jenkinsfile]: This is a no-op, we need a second if in this block: 

```
if (env.run_with_race_detection?.trim()) {
  env.DOCKER_TAG = env.DOCKER_TAG + "-race"
}
```

**Review by nebril — CHANGES_REQUESTED** [review]
Thanks for taking a stab at this! There are some issues with this change, let me know if any of my comments is unclear.

**Review by nebril — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
merging since this is only modifying jenkinsfiles.

**Pull request merged**